### PR TITLE
Add vault kube auth option per instance

### PR DESCRIPTION
**Pull request by dwelch0**
APPSRE-7074

Enables [Vault kubernetes auth](https://developer.hashicorp.com/vault/docs/auth/kubernetes) as option for vault-manager to utilize when authenticating to instances.

Notes: 
* kube auth will only be attempted if an instance definition contains the new optional attr `kubeRoleName` AND new env var `KUBE_SA_TOKEN_PATH` is set. If both conditions are not satisfied, the default auth method defined for the instance will be utilized. The env var will only be set within vault-manager deployments.
* the test cases can not adequately validate this functionality. testing will occur within appsres01ue1

dependent on https://github.com/app-sre/qontract-schemas/pull/391

### CODE REVIEW

File: pkg/vault/instances.go [hunk 1]
```diff
@@ -137,26 +148,26 @@ func processInstances(instances []Instance) (map[string]AuthBundle, error) {
 
 // Creates global map of all vault clients defined in a-i
 // This allows reconciliation of multiple vault instances
-func initClients(instanceCreds map[string]AuthBundle, threadPoolSize int) {
-	vaultClients = make(map[string]*api.Client)
-	masterAddress := configureMaster()
+func initClients(instanceCreds map[string]AuthBundle, kubeSATokenPath string, threadPoolSize int) {
+	vaultClients = make(map[string]*api.Client) // THIS IS THE GLOBAL
```
Comment (cubismod) [inline pkg/vault/instances.go:152]: Thanks for adding that

File: pkg/vault/instances.go [hunk 2]
```diff
@@ -165,47 +176,69 @@ func configureMaster() string {
...
 
+func configureKubeAuthClient(client *api.Client, bundle AuthBundle, kubeSATokenPath string) error {
+	kubeAuth, err := kubernetes.NewKubernetesAuth(
+		bundle.KubeRoleName,
+		kubernetes.WithServiceAccountTokenPath(kubeSATokenPath),
+	)
+	if err != nil {
+		return err
+	}
+	authInfo, err := client.Auth().Login(context.TODO(), kubeAuth)
```
Comment (cubismod) [inline pkg/vault/instances.go:221]: Why the usage of this context type here?

Comment (dwelch0) [reply pkg/vault/instances.go]: Because I forgot to circle back and update it :sweat_smile: 

Good catch. Updated to use timeout

**Review by cubismod — APPROVED** [review]
LGTM!

**Pull request merged**